packages/commerce-sdk-react/src/hooks/utils.ts

```diff
@@ -130,3 +130,9 @@ export const clone = <T>(val: T): T => {
     const entries = Object.entries(val).map(([k, v]) => [k, clone(v)])
     return Object.fromEntries(entries) as T
 }
+
+/** get a list of custom key starting with c_**/
+export const getCustomKeys = <T extends object>(obj: T) => {
+    if (typeof obj !== 'object' || obj === null) return []
+    return Object.keys(obj).filter((key: string): key is `c_${string}` => key.startsWith('c_'))
+}
```

Comment:
Is there a chance that the end user controls the value being passed to this function? => no this is a util func that we have to filter out the custom key for api parameter. I don't see why a user would want to control the val using this function.

Reply:
Why did we choose to not to throw an error? Because this func is an internal util, not expose to users. 
I also saw that we do not throw error for non-objects input. E.g clone function 
Let me know if you think otherwise.

Reply:
So in some cases copying code patterns or style is a good thing, when the pattern or style is good. In this case I think not throwing would lead to hard to fix bugs as a function always returns valid output event when invalid input is passed in. 

In general our team doesn't write a lot of code that throws errors because we live in this javascript world where anything goes. But in typescript it's more common and better practice in general.

Reply:
After discussing this issue, it appears that throwing error in getCustomKey makes sense while `clone` func does require so because of different contexts between two functions. I've made the change as suggested.